sdk/instrumentation/google.golang.org/grpc/server.go

```diff
@@ -94,32 +93,26 @@ func wrapHandler(
 
 		setSchemeAttributes(ctx, span)
 
-		reqBody, err := marshalMessageableJSON(req)
-		if dataCaptureConfig.RpcBody.Request.Value &&
-			len(reqBody) > 0 && err == nil {
-			setTruncatedBodyAttribute("request", reqBody, int(dataCaptureConfig.BodyMaxSizeBytes.Value), span)
+		if dataCaptureConfig.RpcMetadata.Request.Value {
```

Comment:
@ryanericson @puneet-traceable @tim-mwangi This reordering was required because we now only have a single method in filter interface and the scenario of blocking based on headers was failing since the only call to filter was done on body first.

I think we should make similar changes to all instrumentations so that our issue of allow rule not taking precedence over blocking rules also will be solved. 
Or we can pass flowvariable to Evaluate method to specify if we're calling for header eval or body eval

Reply:
So libtraceable interface only has one method called `Evaluate` now? Do the conditions that had it split up into various methods no longer apply?

Reply:
Yes, libtraceable interface has only one method `Evaluate` now. 

> Do the conditions that had it split up into various methods no longer apply?
This I'm not really sure. I looked through the code and couldn't find any reason why we had separate calls (maybe for consistency across agents). In both the cases we call the delegate after the body capture. I'll test with an actual app over the weekend.

Reply:
@tim-mwangi I tested with both a `net/http` app and and a `grpc` app, both of them are blocking with and w/o body